internal/ingress/controller/controller.go

```diff
@@ -580,7 +580,7 @@ func (n *NGINXController) getBackendServers(ingresses []*ingress.Ingress) ([]*in
 		for _, server := range servers {
 			for _, location := range server.Locations {
 				if upstream.Name == location.Backend {
-					if len(upstream.Endpoints) == 0 {
+					if len(upstream.Endpoints) == 0 || len(location.CustomHTTPErrors) != 0 {
 						klog.V(3).Infof("Upstream %q has no active Endpoint", upstream.Name)
 						// check if the location contains endpoints and a custom default backend
 						if location.DefaultBackend != nil {
```

Comment:
What do you think of merging everything until https://github.com/kubernetes/ingress-nginx/pull/3730/files#diff-86ac9ff9d75a0c5005c116e766a6127dR590 and extracting them all together into a helper function something called `shouldCreateUpstreamForLocationDefaultBackend`?

Reply:
done, let me know if that's what you had in mind